Marlin/src/inc/Conditionals_post.h

```diff
@@ -1162,4 +1162,27 @@
   #define NOT_A_PIN 0 // For PINS_DEBUGGING
 #endif
 
+// Number of VFAT entries used. Each entry has 13 UTF-16 characters
+#if ENABLED(SCROLL_LONG_FILENAMES)
+  #define MAX_VFAT_ENTRIES (5)
+#else
+  #define MAX_VFAT_ENTRIES (2)
+#endif
+
+// Force SDCARD_SORT_ALPHA to be enabled for Graphical LCD on LPC1768
+// because of a bug in the shared SPI implementation. (See #8122)
+#if defined(TARGET_LPC1768) && ENABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER)
+  #define SDCARD_SORT_ALPHA         // Keeps one directory level in RAM. Changing
+                                    // directory levels still glitches the screen,
+                                    // but the following LCD update cleans it up.
+  #undef SDSORT_LIMIT
+  #undef SDSORT_USES_RAM
+  #undef SDSORT_USES_STACK
+  #undef SDSORT_CACHE_NAMES
+  #define SDSORT_LIMIT       256    // Maximum number of sorted items (10-256). Costs 27 bytes each.
+  #define SDSORT_USES_RAM    true   // Pre-allocate a static array for faster pre-sorting.
+  #define SDSORT_USES_STACK  false  // Prefer the stack for pre-sorting to give back some SRAM. (Negated by next 2 options.)
+  #define SDSORT_CACHE_NAMES true   // Keep sorted items in RAM longer for speedy performance. Most expensive option.
+#endif
```

Comment:
Don't we need to define all of the options for `SDCARD_SORT_ALPHA`? If `SDCARD_SORT_ALPHA` hasn't been enabled in the configuration, then none of the parameters will be defined, which will lead to errors like:

```
Marlin/src/HAL/HAL_LPC1768/../../inc/SanityCheck.h:330:8: error: #error "SDSORT_CACHE_VFATS must be 2 or greater!"
```

In addition, this prevents the user from defining their own options. Perhaps a better solution would be:

```
// Force SDCARD_SORT_ALPHA to be enabled for Graphical LCD on LPC1768  
// because of a bug in the shared SPI implementation. (See #8122)  
#if defined(TARGET_LPC1768) && ENABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER)  && !ENABLED(SDCARD_SORT_ALPHA)
  #define SDCARD_SORT_ALPHA         // Using SORT feature to keep one directory level in RAM  
                                    // When going up/down directory levels the SD card is  
                                    // accessed but the garbage/lines are removed when the  
                                    // LCD updates  
  
  #define SDSORT_LIMIT       256    // Maximum number of sorted items (10-256). Costs 27 bytes each.  
  #define FOLDER_SORTING     -1     // -1=above  0=none  1=below  
  #define SDSORT_GCODE       false  // Allow turning sorting on/off with LCD and M34 g-code.  
  #define SDSORT_USES_RAM    true   // Pre-allocate a static array for faster pre-sorting.  
  #define SDSORT_USES_STACK  false  // Prefer the stack for pre-sorting to give back some SRAM. (Negated by next 2 options.)  
  #define SDSORT_CACHE_NAMES true   // Keep sorted items in RAM longer for speedy performance. Most expensive option.  
  #define SDSORT_DYNAMIC_RAM false  // Use dynamic allocation (within SD menus). Least expensive option. Set SDSORT_LIMIT before use!  
  #define SDSORT_CACHE_VFATS 2      // Maximum number of 13-byte VFAT entries to use for sorting.  
                                    // Note: Only affects SCROLL_LONG_FILENAMES with SDSORT_CACHE_NAMES but not SDSORT_DYNAMIC_RAM.  
  #define MAX_VFAT_ENTRIES SDSORT_CACHE_VFATS  
#endif
```

Reply:
You make a good point. It would thus be better to use a sanity check to force users to enable it themselves.